### PR TITLE
Prefetch while reading rpc_aggregate_part() outputs

### PR DESCRIPTION
**Pull request by borzunov**
Currently, rpc_aggregate_part() does not receive output parts and decompress them in parallel. This PR fixes it.

__Future work:__

- [ ] Implement prefetching outputs in `iterate_protobuf_handler()` instead.

### CODE REVIEW

File: hivemind/averaging/partition.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def __init__(
         peer_fractions: Sequence[float],
         compression_type: Union["CompressionType", Sequence["CompressionType"]] = CompressionType.NONE,
         part_size_bytes: int = DEFAULT_PART_SIZE_BYTES,
-        prefetch: int = 1,
+        prefetch: int = 256,
```
Comment (borzunov) [inline hivemind/averaging/partition.py:36]: (for debugging)

File: hivemind/p2p/p2p_daemon.py [hunk 2]
```diff
@@ -336,6 +336,9 @@ async def _process_stream() -> None:
                         await P2P.send_protobuf(RPCError(message=str(e)), writer)
 
             with closing(writer):
+                writer.transport.set_write_buffer_limits(low=2 ** 30 - 2 ** 18, high=2 ** 30)
```
Comment (borzunov) [inline hivemind/p2p/p2p_daemon.py:339]: (huge limits for debugging)

File: hivemind/averaging/allreduce.py [hunk 3]
```diff
@@ -153,18 +153,22 @@ async def _communicate_with_peer(self, peer_id: PeerID):
...
-                averaged_part_delta = await loop.run_in_executor(None, deserialize_torch_tensor, msg.tensor_part)
+            async for part_index, (averaged_part_delta, part_code) in aenumerate(
+                amap_in_executor(
+                    lambda msg: (deserialize_torch_tensor(msg.tensor_part), msg.code),
+                    stream,
+                    max_prefetch=self.tensor_part_container.prefetch,
+                )
+            ):
+                if last_code is None:
+                    last_code = part_code
```
Comment (justheuristic) [inline hivemind/averaging/allreduce.py:166]: pls revert to code

**Review by justheuristic — APPROVED** [review]

File: hivemind/averaging/partition.py (hunk 1 above)
Comment (justheuristic) [reply hivemind/averaging/partition.py:36]: mb something like 5?

Comment (borzunov) [reply hivemind/averaging/partition.py]: Changed to 5.

**Pull request merged**